### DISCUSSION TITLE
Endpoints with authentication in debug mode

### DISCUSSION THREAD

**Discussion by sylvain-ri** [apiflask/apiflask#504] [Q&A] [answered]
I have some endpoints that require authentication, and haven't found a way to test them through the web interface http://localhost:5000/docs

Is there a convenient way, something like `app.run(debug=True, token=<private_token>)` or `flask run --token` or to have a field for headers in the webpage?

Maybe this could be added to the docs/tutorials ?

Btw great work, it's a very nice library !

**Comment by tkzt** [comment, marked as answer]
A simple example here:

```python
from apiflask import APIFlask, Schema
from apiflask.fields import String

app = APIFlask(__name__)


class WithAuth(Schema):
    token = String(required=True)


@app.post("/permissions")
@app.input(WithAuth, location="headers")
def foo(headers_data):
    print(headers_data)
    return {"err_code": "", "msg": "", "res": None}


app.run()
```

Then visit `http://127.0.0.1:5000/docs`, and **try it out**:

![image](https://github.com/apiflask/apiflask/assets/38272264/5e50cd0a-5b6f-4c6a-960e-f35dbd713239)

You can populate your token here. Click on `execute` btn, what you get in your terminal will be a string looked like `{'token': 'xxxx'}`. 🥂 

And what you may be interested in: [request-locations](https://apiflask.com/request/#request-locations).

**Reply by sylvain-ri** [reply]
thanks, work well 👌
